nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/jdbc/JDBCRowMapper.java

```diff
@@ -1075,10 +1075,12 @@ protected Serializable copyHierRecursive(IdWithTypes source, Serializable parent
             newId = overwriteId;
             idMap.put(source.id, newId);
         }
-        // recurse in children
-        boolean onlyComplex = parentId == null;
-        for (IdWithTypes child : getChildrenIdsWithTypes(source.id, onlyComplex)) {
-            copyHierRecursive(child, newId, null, null, resetVersion, idMap, idToTypes);
+
+        // recurse in children, exclude regular children (in the case of a versionable folderish)
+        boolean excludeRegularChildren = (parentId == null);
+        for (IdWithTypes child : getChildrenIdsWithTypes(source.id, excludeRegularChildren, excludeSpecialChildren)) {
+            // don't exclude regular children when recursing
+            copyHierRecursive(child, newId, null, null, resetVersion, idMap, idToTypes, true);
```

Comment:
I still don't get the comment or why `true` is passed.

We don't exclude regular children in recursive because `excludeRegularChildren` is defined as `parentId == null` and this will never be true in the recursion. So the comment is at best misplaced.

And you pass `true` for `excludeSpecialChildren` and I thought we *didn't* want to exclude special children when recursing. A unit test must be added to trigger and test this part of the code (unless I'm wrong and I misunderstood the logic and there's already a test covering this of course).

Reply:
Indeed the comment is not usefull now it must have remained from the time there was only one flag.
About specials:
The only level where we have (and need to exclude) special children is indeed the first one. For example CommentsRoot, which then contains regular children which are the comments. 
Maybe I got it wrong but I thought there was only one level of special children right under the document. So in the recursion either we pass exclude or not, nothing will be found. I tought it would be more explicit to hardcode excludeSpecialChildren at true then.

I removed the comment and passed the excludeSpecialChildren flag along.

Reply:
Ok for passing it along. Actually we don't have use cases with specials under specials now, but there could be in the future. We'll revisit the code at that point if this occurs and we need to change the logic.